StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/ReadabilityRules/SA1129CodeFixProvider.cs

```diff
@@ -58,12 +58,32 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
         private static SyntaxNode GetReplacementNode(SyntaxNode node)
         {
             var newExpression = (ObjectCreationExpressionSyntax)node;
+            var nameSyntax = newExpression.Type as NameSyntax;
+            var identifierName = NameSyntaxHelpers.ToNormalizedString(nameSyntax);
 
-            return SyntaxFactory.DefaultExpression(newExpression.Type)
+            SyntaxNode replacement = null;
+            if (identifierName.EndsWith(nameof(CancellationToken), System.StringComparison.OrdinalIgnoreCase))
```

Comment:
The `EndsWith` check is also a problem when there is something like `new MyOwnCancellationToken()`. It should be a verify specific check on `CancellationToken`.